File: src/main/java/jp/co/sunarch/apps/idprovider/model/oauth2/code/AuthorizationCodeAccessTokenRequest.java
package jp.co.sunarch.apps.idprovider.model.oauth2.code;

import jp.co.sunarch.apps.idprovider.model.client.ClientId;
import jp.co.sunarch.apps.idprovider.model.oauth2.AuthorizationCode;
import jp.co.sunarch.apps.idprovider.model.oauth2.RedirectUri;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * アクセストークンリクエスト
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AuthorizationCodeAccessTokenRequest {

  private AuthorizationCode code;
  private RedirectUri redirectUrl;
  private ClientId clientId;

}

File: src/main/kotlin/jp/co/sunarch/apps/idprovider/model/oauth2/code/AuthorizationCodeAccessTokenRequest.kt
package jp.co.sunarch.apps.idprovider.model.oauth2.code

import jp.co.sunarch.apps.idprovider.model.client.ClientId
import jp.co.sunarch.apps.idprovider.model.oauth2.AuthorizationCode
import jp.co.sunarch.apps.idprovider.model.oauth2.RedirectUri

/**
 * アクセストークンリクエスト
 */
class AuthorizationCodeAccessTokenRequest {

  private val code: AuthorizationCode? = null
  private val redirectUrl: RedirectUri? = null
  private val clientId: ClientId? = null

}
